t/role/record_nonblocking.t

```diff
@@ -0,0 +1,49 @@
+use Mojo::Base -strict;
+use Mojo::File qw(tempdir);
+use Mojo::UserAgent;
+use Mojo::UserAgent::Mockable::Serializer;
+use Test::More;
+use Test::Mojo;
+
+my $t = Test::Mojo->new;
+my (@results, @transactions);
+my @promises = (
+  $t->ua->get_p(q{https://www.vam.ac.uk/api/json/museumobject/O1}),
```

Comment:
This is fine, but in future I think we should do away with the external websites altogether and instead fork a dedicated HTTP server to test against. Provided, of course, that doesn't break Jenkins...